pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -203,3 +204,150 @@ class Meta:
+                        self.name, self.stream, self.version,
+                        self.context, self.arch)
+
+        return relative_path
+
+    @classmethod
+    async def pre_migrate_content_detail(cls, content_batch):
+        """
+        Pre-migrate Pulp 2 modulemd content with all the fields needed to create a Pulp 3 content.
+
+        Args:
+             content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
+
+        """
+        pulp2_id_obj_map = {pulp2content.pulp2_id: pulp2content for pulp2content in content_batch}
+        pulp2_ids = pulp2_id_obj_map.keys()
+        pulp2_content_batch = pulp2_models.Modulemd.objects.filter(id__in=pulp2_ids)
+        pulp2modules_to_save = [
+            cls(name=md.name,
+                stream=md.stream,
+                version=md.version,
+                context=md.context,
+                arch=md.arch,
+                dependencies=md.dependencies,
+                artifacts=md.artifacts,
+                checksum=md.checksum,
+                pulp2content=pulp2_id_obj_map[md.id])
+            for md in pulp2_content_batch]
+        cls.objects.bulk_create(pulp2modules_to_save, ignore_conflicts=True)
+
+    async def create_pulp3_content(self):
+        """
+        Create a Pulp 3 Module content for saving it later in a bulk operation.
+        """
+        return Modulemd(name=self.name, stream=self.stream, version=self.version,
+                        context=self.context, arch=self.arch, artifacts=self.artifacts,
+                        dependencies=self.dependencies)
+
+
+class Pulp2ModulemdDefaults(Pulp2to3Content):
+    """
+    Pulp 2to3 detail content model to store Pulp2 ModulemdDefaults content details.
+    """
+
+    # Unit key fields
+    module = models.TextField()
+    stream = models.TextField()
+    profiles = JSONField(dict)
+    digest = models.TextField()
+    repo_id = models.TextField()
+
+    pulp2_type = 'modulemd_defaults'
+
+    class Meta:
+        unique_together = ('digest', 'repo_id', 'pulp2content')
+        default_related_name = 'modulemd_defaults_detail_model'
+
+    @property
+    def expected_digests(self):
+        """Return expected digests."""
+        return {'sha256': self.digest}
+
+    @property
+    def expected_size(self):
+        """Return expected size."""
+        return
+
+    @property
+    def relative_path_for_content_artifact(self):
+        """Return relative path."""
+        relative_path = '{}{}snippet'.format(self.module, self.stream)
+
+        return relative_path
+
+    @classmethod
+    async def pre_migrate_content_detail(cls, content_batch):
+        """
+        Pre-migrate Pulp 2 defaults content with all the fields needed to create a Pulp 3 content.
+
+        Args:
+             content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
+
+        """
+
+        def _get_profiles(profiles):
+
+            bson_string = BSON(profiles, encoding='utf8')
```

Comment:
Could you elaborate why it is needed?

Reply:
unfortunately this is how we store profies in pulp2
```
"profiles" : "9\u0000\u0000\u0000\u00040.71\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000flipper\u0000\u0000\u00045.21\u0000\u0014\u0000\u0000\u0000\u00020\u0000\b\u0000\u0000\u0000default\u0000\u0000\u0000",
```
i need to create a bson string so i can normally decode it and work with the data.